src/include/gcc.h

```diff
@@ -157,6 +176,29 @@ WT_ATOMIC_FUNC(i64, int64_t, int64_t *vp, int64_t v)
 WT_ATOMIC_FUNC(iv64, int64_t, volatile int64_t *vp, volatile int64_t v)
 WT_ATOMIC_FUNC(size, size_t, size_t *vp, size_t v)
 
+/*
+ * We can't use the WT_ATOMIC_FUNC macro for booleans as __atomic_add_fetch and __atomic_sub_fetch
+ * don't accept booleans when compiling with clang. Define them individually.
+ */
+static inline bool
+__wt_atomic_loadbool(bool *vp)
```

Comment:
I'm surprised these don't trip the `s_style` script `missing or malformed comment` checks. Given that these are actual functions we should have something like what we have for the cas function.
```
/*
 * __wt_atomic_cas_ptr --
 *     Pointer compare and swap.
 */
```

Reply:
Added in [8338d6c](https://github.com/wiredtiger/wiredtiger/pull/10236/commits/8338d6c2b42374421e4114c8d9856085d665932c)